core/src/main/scala/org/apache/spark/ui/exec/ExecutorThreadDumpPage.scala

```diff
@@ -48,7 +49,13 @@ private[ui] class ExecutorThreadDumpPage(parent: ExecutorsTab) extends WebUIPage
     val time = System.currentTimeMillis()
     val maybeThreadDump = sc.get.getExecutorThreadDump(executorId)
 
-    val content = maybeThreadDump.map { threadDump =>
+    val grepExp = Option(request.getParameter("grepexp")).map(Pattern.compile(_, Pattern.MULTILINE))
+
+    val filteredContent = maybeThreadDump.map { threadDump =>
+      threadDump.filter(thread =>
+        grepExp.map { grep => grep.matcher(thread.stackTrace).find()}.getOrElse(true)
+      )
```

Comment:
also minor: what do you think about renaming `threadDump` to `threadDumps` (plural)?  again not a big deal, and I know it was already `threadDump` in this file, not something you added ... but the singular always confuses me for just a second, I think that we're filtering individual elements of the stack trace, not complete threads.